GW2EIEvtcParser/EIData/ProfHelpers/Mesmer/ChronomancerHelper.cs

```diff
@@ -84,5 +86,173 @@ internal static bool IsKnownMinionID(int id)
+                var skill = new SkillModeDescriptor(player, Spec.Chronomancer, WellOfEternity, SkillModeCategory.Heal);
+                foreach (EffectEvent effect in wellsOfEternity)
+                {
+                    (int, int) lifespan = ProfHelper.ComputeEffectLifespan(log, effect, 5000);
+                    var connector = new PositionConnector(effect.Position);
+                    replay.Decorations.Add(new CircleDecoration(false, 0, 240, lifespan, color.WithAlpha(0.5f).ToString(), connector).UsingSkillMode(skill));
+                    replay.Decorations.Add(new IconDecoration(ParserIcons.EffectWellOfEternity, CombatReplaySkillDefaultSizeInPixel, CombatReplaySkillDefaultSizeInWorld, 0.5f, lifespan, connector).UsingSkillMode(skill));
+                }
+            }
+            // Well of Eternity - Pulses
+            if (log.CombatData.TryGetEffectEventsBySrcWithGUIDs(player.AgentItem, new string[] { EffectGUIDs.ChronomancerWellOfEternityPulse, EffectGUIDs.ChronomancerWellOfEternityExplosion }, out IReadOnlyList<EffectEvent> wellsOfEternityPulses))
+            {
+                var skill = new SkillModeDescriptor(player, Spec.Chronomancer, WellOfEternity, SkillModeCategory.Heal);
+                foreach (EffectEvent effect in wellsOfEternityPulses)
+                {
+                    int effectTimeStart = (int)effect.Time;
+                    int effectTimeEnd = effectTimeStart + 1000;
+                    var connector = new PositionConnector(effect.Position);
+                    replay.Decorations.Add(new CircleDecoration(false, effectTimeEnd, 240, (effectTimeStart, effectTimeEnd), color.WithAlpha(0.5f).ToString(), connector).UsingSkillMode(skill));
+                }
+            }
+
+            // Well of Action
+            if (log.CombatData.TryGetEffectEventsBySrcWithGUID(player.AgentItem, EffectGUIDs.ChronomancerWellOfAction, out IReadOnlyList<EffectEvent> wellsOfAction))
+            {
+                var skill = new SkillModeDescriptor(player, Spec.Chronomancer, WellOfAction);
+                foreach (EffectEvent effect in wellsOfAction)
+                {
+                    (int, int) lifespan = ProfHelper.ComputeEffectLifespan(log, effect, 5000);
+                    var connector = new PositionConnector(effect.Position);
+                    replay.Decorations.Add(new CircleDecoration(false, 0, 240, lifespan, color.WithAlpha(0.5f).ToString(), connector).UsingSkillMode(skill));
+                    replay.Decorations.Add(new IconDecoration(ParserIcons.EffectWellOfAction, CombatReplaySkillDefaultSizeInPixel, CombatReplaySkillDefaultSizeInWorld, 0.5f, lifespan, connector).UsingSkillMode(skill));
+                    
+                    // Well pulses - Hard coded because the effects don't have a Src
+                    int pulseTimeDelay = 0;
+                    for (int i = 0; i < 4; i++)
+                    {
+                        int effectTimeStart = (int)effect.Time + pulseTimeDelay;
```

Comment:
As this is hardcoded, the for loop should break if effectTimeStart > lifespan.item2.